src/widget/style.cpp

```diff
@@ -219,17 +219,17 @@ QFont Style::getFont(Font font)
 
     static int defSize = QFontInfo(QFont()).pixelSize();
 
-    static QFont fonts[] = {
-        appFont(defSize + 3, QFont::Bold),   // extra big
-        appFont(defSize + 1, QFont::Normal), // big
-        appFont(defSize + 1, QFont::Bold),   // big bold
-        appFont(defSize, QFont::Normal),     // medium
-        appFont(defSize, QFont::Bold),       // medium bold
-        appFont(defSize - 1, QFont::Normal), // small
-        appFont(defSize - 1, QFont::Light),  // small light
+    static std::map<Font, QFont> fonts = {
+        {Font::ExtraBig, appFont(defSize + 3, QFont::Bold)},
+        {Font::Big, appFont(defSize + 1, QFont::Normal)},
+        {Font::BigBold, appFont(defSize + 1, QFont::Bold)},
+        {Font::Medium, appFont(defSize, QFont::Normal)},
+        {Font::MediumBold, appFont(defSize, QFont::Bold)},
+        {Font::Small, appFont(defSize - 1, QFont::Normal)},
+        {Font::SmallLight, appFont(defSize - 1, QFont::Light)},
     };
 
-    return fonts[static_cast<int>(font)];
+    return fonts[font];
```

Comment:
`.at(font)`? `operator[]` will create a new element and return that if the key doesn't exist in the map. :P `.at()` will try to throw and explode instead.

Reply:
I like exploding over UB, done :D And moved to const.